docs/building_decision_trees.rst

```diff
@@ -0,0 +1,246 @@
+########################################################
+Understanding and building a component selection process
+########################################################
+
+``tedana`` involves transforming data into components via ICA, and then calculating metrics for each component.
+Each metric has one value per component that is stored in a comptable or component_table dataframe. This structure
+is then passed to a "decision tree" which through a series of binary choices categories each component as **accepted** or
+**rejected**. The time series for the rejected components are regressed from the data in the final denoising step.
```

Comment:
I wonder if we should provide more conservative approaches to denoise the data. See [Moia et al., 2021](https://www.sciencedirect.com/science/article/pii/S1053811921001919).

TL;DR: regressing out the rejected components without orthogonalizing them to the accepted components can be too aggressive.

Reply:
I'm hesitant to drop the denoised data completely, but I agree that users' general workflow should probably be to get out the components and perform the regression separately. That way, they can include other regressors (e.g., motion regressors) and orthogonalize as needed afterwards. I'm not sure if orthogonalizing w.r.t. the accepted components should happen within tedana via the `tedort` option or afterwards with the other regressors included though.

Reply:
Regarding [Moia et al., 2021](https://www.sciencedirect.com/science/article/pii/S1053811921001919), I would emphasize that that's a very special experimental condition, such as cerebrovascular reactivity, where the task produces a global, widespread response and multiple task-related components were classified as non-BOLD by TEDANA. Therefore, we had to do a manual classification, and even within the noisy components we observed substantial correlation with the task because breath-holding respiratory and motion effects are highly correlated with the task.

Reply:
This is a pretty big workflow change, I'd like to separate it from the decision tree modularization just because that's already really big. Could we discuss this in an issue?

Reply:
Yes, this seems more like a separate documentation issue than something to be dealt with in this PR. #823 is probably a good place to follow up.